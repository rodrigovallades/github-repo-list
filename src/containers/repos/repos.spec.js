import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import { Repos } from './index'

describe('Repos container', function () {
  beforeEach(function() {
    global.sessionStorage = jest.genMockFunction();
    global.sessionStorage.setItem = jest.genMockFunction();
    global.sessionStorage.getItem = jest.genMockFunction();
  });

  it('renders "Repos" deep correctly', () => {
    const tree = renderer.create(<Repos/>);
    expect(tree).toMatchSnapshot();
  });

  it('renders "Repos" shallow correctly', () => {
    const component = shallow(<Repos/>);
    expect(component.exists()).toEqual(true);
  });
})
