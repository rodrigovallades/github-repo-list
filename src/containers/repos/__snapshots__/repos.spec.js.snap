// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repos container renders "Repos" deep correctly 1`] = `
<div>
  <div
    className="container"
  >
    <h1
      className="title"
    >
      <span
        className="badge badge-light"
      >
        0
      </span>
       Repositories
    </h1>
    <input
      className="filter form-control"
      onChange={[Function]}
      placeholder="Filter repository"
      type="text"
      value=""
    />
    <div
      className="repos"
    >
      <p>
        No repos found in this account.
      </p>
    </div>
    <div
      className="alert alert-info"
      role="alert"
    >
      <span
        className="octicon octicon-info"
      />
       Tip: Download the 
      <strong>
        <a
          href="https://github.com/zalmoxisus/redux-devtools-extension"
          target="_blnk"
        >
          Redux DevTools
        </a>
      </strong>
       to inspect the Redux store state.
    </div>
  </div>
</div>
`;
